src/convertToObject.js

```diff
@@ -3,16 +3,28 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example
+ * in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS
+ * properties (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .filter(style => style.trim())
+    .map(style => {
+      const [property, value] = style.split(':').map(part => part.trim());
+
+      return { [property]: value };
+    })
+    .reduce((result, style) => ({
```

Comment:
good decision to use `reduce`, but with `reduce` you can simplify your code and get rid of `map` inside `map`, like 

> return sourceString.split(";").reduce((result, style) => {
>     const stylePair = style.split(":");
>     const key = stylePair[0]?.trim();
>     const value = stylePair[1]?.trim();
>     if (key && value) {
>       result[key] = value;
>     }
>     return result;
>   }, {});